src/irmin/contents.ml

```diff
@@ -193,6 +193,16 @@ end
 module String = struct
```

Comment:
Perhaps we should provide e.g. `Contents.String_v1` with the old pre-hash semantics, to ease migration for Irmin 2.0 users?

Reply:
I exposed a `V2` for contents, nodes and commits (as I had to change the prehash for the default implementation of all 3 stores). This was a nice suggestion, because I can use the v2 versions to test the existing stores (and avoid regenerating them). The rest of the tests use the default implementations, which is now without prehash collisions.